File: STRIVER_SDE_SHEET/Recursion/PalindromicPartitioning.py
def helper(ind,s,curr,ans):
        if(ind==len(s)):
            ans.append(curr[:])
        for i in range(ind,len(s)):
            if(s[ind:i+1]==s[ind:i+1][::-1]):
                curr.append(s[ind:i+1])
                helper(i+1,s,curr,ans)
                curr.pop()
    
    
ans = []
curr = []
s = input()
helper(0,s,curr,ans)
for i in ans:
    print(i)

        

File: STRIVER_SDE_SHEET/Recursion/PalindromicPartitioning.java
package Recursion;

import java.util.ArrayList;
import java.util.Scanner;

public class PalindromicPartitioning {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        ArrayList<ArrayList<String>> ans = new ArrayList<>();
        in.close();
        palindromePart(0,s,new ArrayList<String>(),ans);
        for(ArrayList<String> al : ans){
            for(String str : al){
                System.out.print(str+" ");
            }
            System.out.println();
        }
    }

    private static void palindromePart(int ind, String s, ArrayList<String> curr, ArrayList<ArrayList<String>> ans) {
        if (ind == s.length()) {
            ans.add(new ArrayList<>(curr));
        }
        for (int i = ind; i < s.length(); i++) {
            if (isPal(s, ind, i)) {
                curr.add(s.substring(ind, i + 1));
                palindromePart(i + 1, s, curr, ans);
                curr.remove(curr.size() - 1);
            }
        }
    }

    static boolean isPal(String s, int st, int nd) {
        while (st < nd) {
            if (s.charAt(st++) != s.charAt(nd--))
                return false;
        }
        return true;
    }
}
